include/crocoddyl/multibody/costs/contact-cop-position.hpp

```diff
@@ -0,0 +1,165 @@
+// BSD 3-Clause License
+//
+// Copyright (C) 2020, University of Duisburg-Essen, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MULTIBODY_COSTS_CONTACT_COP_POSITION_HPP_
+#define CROCODDYL_MULTIBODY_COSTS_CONTACT_COP_POSITION_HPP_
+
+#include "crocoddyl/multibody/fwd.hpp"
+#include "crocoddyl/multibody/cost-base.hpp"
+#include "crocoddyl/multibody/contact-base.hpp"
+#include "crocoddyl/multibody/contacts/contact-3d.hpp"
+#include "crocoddyl/multibody/contacts/contact-6d.hpp"
+#include "crocoddyl/multibody/data/contacts.hpp"
+#include "crocoddyl/multibody/frames.hpp"
+#include "crocoddyl/multibody/data/multibody.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class CostModelContactCoPPositionTpl : public CostModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef CostModelAbstractTpl<Scalar> Base;
+  typedef CostDataContactCoPPositionTpl<Scalar> Data;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
+  typedef CostDataAbstractTpl<Scalar> CostDataAbstract;
+  typedef ActivationModelAbstractTpl<Scalar> ActivationModelAbstract;
+  typedef ActivationModelQuadTpl<Scalar> ActivationModelQuad;
+  typedef DataCollectorAbstractTpl<Scalar> DataCollectorAbstract;
+  typedef typename MathBase::Vector2s Vector2s;
+  typedef typename MathBase::Vector3s Vector3s;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+  typedef typename MathBase::MatrixX3s MatrixX3s;
+  typedef typename MathBase::Matrix46s Matrix46s;
+
+  CostModelContactCoPPositionTpl(boost::shared_ptr<StateMultibody> state,
+                          boost::shared_ptr<ActivationModelAbstract> activation, const std::size_t& frame_id, 
+                          const Vector2s& cop_region, const Vector3s& normal, const std::size_t& nu);
+  virtual ~CostModelContactCoPPositionTpl();
+
+  virtual void calc(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                    const Eigen::Ref<const VectorXs>& u);
+  virtual void calcDiff(const boost::shared_ptr<CostDataAbstract>& data, const Eigen::Ref<const VectorXs>& x,
+                        const Eigen::Ref<const VectorXs>& u);
+  virtual boost::shared_ptr<CostDataAbstract> createData(DataCollectorAbstract* const data);
+
+  const std::size_t& get_frame_id() const;
+  const Vector2s& get_cop_region() const;
+  const Vector3s& get_normal() const;
+
+ protected:
+  using Base::activation_;
+  using Base::nu_;
+  using Base::state_;
+  using Base::unone_;
+
+ private: 
+  const std::size_t frame_id_; //!< contact frame id
+  const Vector2s cop_region_; //!< desired 2d area (length, width) of the CoP defined w.r.t. the contact frame
```

Comment:
Name it support_area

Reply:
Also considered this. However, wouldn't it be a bit confusing since the ideal CoP region can be a subregion of the actual support area of the foot?

Reply:
I've changed the name to `support_region` as suggested by Carlos.